specification/communication/data-plane/CallAutomation/stable/2023-03-06/communicationservicescallautomation.json

```diff
@@ -0,0 +1,2155 @@
+      "required": [
+        "incomingCallContext"
+      ],
+      "type": "object",
+      "properties": {
+        "incomingCallContext": {
+          "description": "The context associated with the call.",
+          "type": "string"
+        },
+        "callRejectReason": {
+          "$ref": "#/definitions/CallRejectReason"
+        }
+      }
+    },
+    "TransferToParticipantRequest": {
+      "description": "The request payload for transferring call to a participant.",
+      "required": [
+        "targetParticipant"
+      ],
+      "type": "object",
+      "properties": {
+        "targetParticipant": {
+          "$ref": "#/definitions/CommunicationIdentifierModel",
+          "description": "The identity of the target where call should be transferred to."
+        },
+        "operationContext": {
+          "description": "Used by customers when calling mid-call actions to correlate the request to the response event.",
+          "type": "string"
+        }
+      }
+    },
+    "TransferCallResponse": {
+      "description": "The response payload for transferring the call.",
+      "type": "object",
+      "properties": {
+        "operationContext": {
+          "description": "The operation context provided by client.",
+          "type": "string"
+        }
+      }
+    },
+    "PlayRequest": {
+      "required": [
+        "playSourceInfo"
+      ],
+      "type": "object",
+      "properties": {
+        "playSourceInfo": {
+          "$ref": "#/definitions/PlaySource",
+          "description": "The source of the audio to be played."
+        },
+        "playTo": {
+          "description": "The list of call participants play provided audio to.\r\nPlays to everyone in the call when not provided.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationIdentifierModel"
+          }
+        },
+        "playOptions": {
+          "$ref": "#/definitions/PlayOptions",
+          "description": "Defines options for playing the audio."
+        },
+        "operationContext": {
+          "description": "The value to identify context of the operation.",
+          "type": "string"
+        }
+      }
+    },
+    "PlaySourceType": {
+      "description": "Defines the type of the play source",
+      "enum": [
+        "file"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "PlaySourceType",
+        "modelAsString": true
+      }
+    },
+    "PlaySource": {
+      "required": [
+        "sourceType"
+      ],
+      "type": "object",
+      "properties": {
+        "sourceType": {
+          "$ref": "#/definitions/PlaySourceType"
+        },
+        "playSourceId": {
+          "description": "Defines the identifier to be used for caching related media",
+          "type": "string"
+        },
+        "fileSource": {
+          "$ref": "#/definitions/FileSource",
+          "description": "Defines the file source info to be used for play"
+        }
```

Comment:
how will this PlaySource grow, if there is more sourceType added in future?

Reply:
Yes - there will be few source type planned ahead (TextSource, SSMLSource)

Reply:
The ask is try see how you grow this model, if you have more types. And check (yourself, or with team) whether the change is easy to use for customer and also not a breaking change.

Reply:
We would add more types under PlacySourceType, and actual payload to be addition inside of PlaySource's properties.
So if were to add TextSource and ssmlSource,

    "PlaySource": {
      "required": [
        "sourceType"
      ],
      "type": "object",
      "properties": {
        "sourceType": {
          "$ref": "#/definitions/PlaySourceType"
        },
        "playSourceId": {
          "description": "Defines the identifier to be used for caching related media",
          "type": "string"
        },
        "fileSource": {
          "$ref": "#/definitions/FileSource",
          "description": "Defines the file source info to be used for play"
        },
        "textSource": {
          "$ref": "#/definitions/TextSource",
          "description": "Defines the text source info to be used for play"
        },
        "ssmlSource": {
          "$ref": "#/definitions/SsmlSource",
          "description": "Defines the ssml(Speech Synthesis Markup Language) source info to be used for play"
        }
      }
    }
    
 Customer would define PlaySourceType, and then fill out the proper source based on the type. This works very similar to Communication Service's CommunicationIdentifier, so the Communication API developer should be familiar with this approach.

Reply:
Jeffrey is working on a new polymorphism guideline which proposes to do what the CommunicationIdentifier is doing: the discriminator would be called kind, and each kind value is literally the property name that gets populated. So it would like

```json
"PlaySource": {
  "required": [
    "sourceType"
  ],
  "type": "object",
  "properties": {
    "kind": {
      "$ref": "#/definitions/PlaySourceType" // this is 'file' | 'text' | 'ssml'
    },
    "playSourceId": {
      "description": "Defines the identifier to be used for caching related media",
      "type": "string"
    },
    "file": {
      "$ref": "#/definitions/FileSource",
      "description": "Defines the file source info to be used for play"
    },
    "text": {
      "$ref": "#/definitions/TextSource",
      "description": "Defines the text source info to be used for play"
    },
    "ssml": {
      "$ref": "#/definitions/SsmlSource",
      "description": "Defines the ssml(Speech Synthesis Markup Language) source info to be used for play"
    }
  }
}
```

Reply:
If we follow this style of polymorphism, we should do it everywhere

Reply:
Thank you  @DominikMe - I will make those proper update in API with proper naming.

sourceType -> kind
fileSource -> file